### ISSUE TITLE
Impossible to pass --require-path to pulp browserify

### ISSUE THREAD

**Issue by hdgarrood** [purescript-contrib/pulp#174]
See https://github.com/purescript-contrib/purescript-react/issues/58

Because the remainder of the arguments with `pulp browserify` go to `psc-bundle` and not `psc`, it's impossible to pass `--require-path=../` when `pulp browserify` calls `psc`. This means that you can't just run `pulp browserify --to bundle.js --require-path=../`, you have to do something like running `pulp build --require-path=../ && pulp browserify --require-path=../ --to bundle.js`, and a straight `pulp browserify --to bundle.js` breaks.

We might wait for https://github.com/purescript/purescript/issues/1836 to be fixed before addressing this.

The other thing about remainder arguments is that it makes it quite easy to make mistakes without realising, because unrecognised arguments just get passed along. If you misspell a pulp argument, I think it would be better to receive an error. What do you think about removing them and adding options for extra arguments for specific executables?

Eg,

```
pulp build --psc-args="--require-path=../"
```

or

```
pulp browserify --psc-args="--require-path=../" --psc-bundle-args="--require-path=../"
```

**Comment by nwolverson** [comment]
Requiring extra arguments to be quoted does complicate things, eg when the entire command is quoted (eg in npm run script) or you want to pass quoted arguments (maybe that never happens for psc?). The explicit separator style feels a little lighter (eg everything after `--`) but I guess that's a problem with 2 sets of extra-arguments.

That said remainder arguments with pulp -> psa -> psc and nonspecific error message from psc were quite confusing.

**Comment by hdgarrood** [comment]
Yeah. It seems like this situation is going to change fairly soon, though, so I think we should revisit this after purescript/purescript#1836 and purescript/purescript#1838 are fixed.

**Comment by natefaubion** [comment]
This is also a problem if people are using `psa`. The extra args to control warnings go to `psc-bundle` instead of `psa`, and obviously trigger an error.

**Comment by hdgarrood** [comment]
I think reinstating the `--skip-compile` flag (I think I accidentally removed it during the port) would mostly fix this, or at least provide a decent workaround.

**Comment by hdgarrood** [comment]
Resolved by #232.
